src/components/hoc/withAdapter.jsx

```diff
@@ -0,0 +1,71 @@
+import React, {useEffect, useState} from 'react';
+
+import WebexDataProvider from '../WebexDataProvider/WebexDataProvider';
+
+/**
+ * withAdapter is a high-order component that takes a component (WrappedComponent)
+ * and returns a new component (WithAdapter) that, given an adapter, takes care of
+ * connecting/disconnecting this adapter instance and providing in on a react context.
+ *
+ * Example:
```

Comment:
Using [JSDoc tag](https://jsdoc.app/tags-example.html)
```suggestion
 * @example
```